src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,245 @@
+from tqec.circuit.schedule import ScheduledCircuit
+from tqec.circuit.qubit_map import QubitMap
+from tqec.plaquette.plaquette import Plaquette
+from tqec.plaquette.qubit import PlaquetteQubits
+from tqec.plaquette.qubit import SquarePlaquetteQubits
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from stim import Circuit as stim_Circuit
+
+
+class BasisEnum(Enum):
+    X = 'x'
+    Y = 'y'
+    Z = 'z'
+
+
+class ExtendedBasisEnum(Enum):
+    X = BasisEnum.X.value
+    Y = BasisEnum.Y.value
+    Z = BasisEnum.Z.value
+    H = 'h'
+
+
+@dataclass
+class RPNG:
+    """Organize a single RPNG value
+
+    ----
+    Simplified RPNG format:
+
+    -z1-
+    rpng
+    
+    (r) data qubit reset basis or h or -  
+    (p) data basis for the controlled operation (x means CNOT controlled on the ancilla and targeting the data qubit, y means CY, z means CZ)  
+    (n) time step (positive integers, all distinct, typically in 1-5)  
+    (g) data qubit measure basis or h or -
+
+    Assumptions on the circuit:
+    - if not otherwise stated, a basis can be {x,y,z}
+    - the ancilla is always the control qubit for the CNOT, CY, or CZ gates
+    - time step of r same as ancilla reset (always 0)
+    - time step of g same as ancilla measurement (by default 6)
+
+    ----
+    Extended RPNG format:
+
+    The extra field (a) is used together with (p) to determine the 2Q gate involving the
+    ancilla and data qubits. For example:
+    - a=z, p=x --> CNOT controlled by the ancilla and targeting the data
+    - a=z, p=z --> CZ   controlled by the ancilla and targeting the data
+    - a=y, p=z --> CY   controlled by the data    and targeting the ancilla
+    By default a=z.
+    """
+    r: Optional[ExtendedBasisEnum]
```

Comment:
Can't remember what we settled on, but I think the `h` entry was initially only on the measurement part. I just wanted to leave that as a note, I personally prefer to have it on both measurement and reset.

Reply:
I don't mind leaving the user more freedom to define plaquettes.
The only reason to remove the option seems to be to avoid mistyping the RPNG string (without being flagged by the RPNG validation).